File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cs
public class Solution {
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {
        int maximum= candies.Max();
        bool[] ans = new bool[candies.Length];
        for ( int i =0;i<candies.Length;i++){
            if (candies[i]+extraCandies >= maximum){
                ans[i]=true ;
            }
            else {
                ans[i]= false;
            }


        }
        return ans ;
        
    }
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int largest = candies[0];
        for (  int i =0;i<candies.length ;i++){
            if (candies[i]>largest){
                largest = candies[i];
            }
        }
        List<Boolean> ans = new ArrayList<>(candies.length);
        for ( int i=0;i<candies.length ;i++){
            if (candies[i]+extraCandies>=largest){
                ans.add(true) ;
            }
            else{
                ans.add( false) ;
            }
        }
        return ans ;
    }
}